lib/spack/spack/relocate.py

```diff
@@ -202,7 +192,9 @@ def get_filetype(path_name):
     """
     file = Executable('file')
     file.add_default_env('LC_ALL', 'C')
-    output = file('-b', '-h', '%s' % path_name,
+    file.add_default_arg('-b')
```

Comment:
If I understand correctly, this change is not required and it was actually your commit https://github.com/spack/spack/pull/6140/commits/aaee131663a89fb28c242ad650572bedea98b90d which fixes the issue this was intended to address. (the old and new logic in this function should be equivalent anyhow).

If I got that right, could you change it back? Otherwise let me know. This is my only remaining issue.